overlord/devicestate/devicemgr.go

```diff
@@ -1025,7 +1028,17 @@ func (m *DeviceManager) Reboot(systemLabel, mode string) error {
 		systemLabel = currentSys.System
 	}
 
-	return m.doSystemAction(systemLabel, SystemAction{Mode: mode})
+	err := m.doSystemAction(systemLabel, SystemAction{Mode: mode})
+	// system already in the requested mode, just reboot
+	if err == ErrAlreadyInRequestedMode {
+		logger.Noticef("restarting system")
```

Comment:
we have the same code written already once in the function, maybe this should be a small helper,  also I prefer the defer style of the two